app/decorators/cart_decorator.rb

```diff
@@ -4,6 +4,8 @@ class CartDecorator < Draper::Decorator
   def total_price
     if self.getProp('origin') == 'ncr'
       self.getProp('amount').to_f
+    elsif self.getProp('origin') == 'gsa18f'
+      self.getProp('cost_per_unit').to_f * self.getProp('quantity').to_f
     else
```

Comment:
Mind switching to a `case` statement?

http://ruby-doc.org/docs/keywords/1.9/Object.html#method-i-case